### PR TITLE
Adding SQL AutoDR support

### PR DESCRIPTION
**Pull request by franktsa**
Adding support for AutoDr functionality, which is a configuration object under the server. Operations include:
•Create (PUT)
•Read (GET)
•Update / Failover (POST)
•Delete (DELETE)

CR: http://codeflow/dashboard/review/display/franktsa-e5e79d5fd13142c6b8acc29d8e25e34d

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@franktsa__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (franktsa). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by stankovski** [comment]
@azuresdkci add to whitelist

File: src/ResourceManagement/Sql/Sql.Tests/ScenarioTests/Sql2ScenarioHelper.cs [hunk 1]
```diff
@@ -25,11 +25,13 @@ namespace Sql2.Tests.ScenarioTests
 {
     class Sql2ScenarioHelper
     {
-        
+
         /// <summary>
         /// The region in which the tests will create their needed resources
         /// </summary>
-        private static string TestEnvironmentRegion = "Australia East";
+        private static string TestEnvironmentRegion = "North Europe";
```
Comment (markcowl) [inline src/ResourceManagement/Sql/Sql.Tests/ScenarioTests/Sql2ScenarioHelper.cs]: Please update this so you are not simply using hard-coded regions.  These regiosn may or may not be available depending on which environment or test cluster is being targeted.

File: src/ResourceManagement/Sql/Sql.Tests/ScenarioTests/Sql2.ServerDisasterRecoveryConfigurationTests.cs [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+                    "12.0",
+                    true,
+                    (sqlClient, resGroupName, server1, server2) =>
+                    {
+                        validateDrcNotExist(sqlClient, resGroupName, server1.Name, failoverAliasName);
+                        validateDrcNotExist(sqlClient, resGroupName, server2.Name, failoverAliasName);
+
+                        // Create and verify
+                        //
+                        ServerDisasterRecoveryConfigurationCreateOrUpdateResponse createResponse = CreateDrc(sqlClient, resGroupName, server1, server2, failoverAliasName);
```
Comment (markcowl) [inline src/ResourceManagement/Sql/Sql.Tests/ScenarioTests/Sql2.ServerDisasterRecoveryConfigurationTests.cs]: @franktsa - line breaks - please break lines ~80 columns

**Pull request merged**